opensrp-anc/src/main/java/org/smartregister/anc/library/listener/ContactTaskDisplayClickListener.java

```diff
@@ -0,0 +1,227 @@
+     */
+    private JSONArray getExpansionPanelValues(JSONObject taskValue, String taskKey) {
+        JSONArray values = new JSONArray();
+        if (taskValue != null && StringUtils.isNotBlank(taskKey)) {
+            JSONArray taskValueArray = new JSONArray();
+            taskValueArray.put(taskValue);
+            values = contactJsonFormUtils.loadExpansionPanelValues(taskValueArray, taskKey);
+        }
+
+        return values;
+    }
+
+    /**
+     * Returns a map of the expansion panel values
+     *
+     * @param secondaryValues {@link JSONArray}
+     * @return expansionPanelValuesMap = {@link Map}
+     */
+    private Map<String, ExpansionPanelValuesModel> getSecondaryValues(JSONArray secondaryValues) {
+        Map<String, ExpansionPanelValuesModel> stringExpansionPanelValuesModelMap = new HashMap<>();
+        if (secondaryValues != null && secondaryValues.length() > 0) {
+            stringExpansionPanelValuesModelMap = contactJsonFormUtils.createSecondaryValuesMap(secondaryValues);
+        }
+        return stringExpansionPanelValuesModelMap;
+    }
+
+    /**
+     * Loads the sub forms using the name on the accordion.  It returns the sub form fields
+     *
+     * @param taskValue {@link JSONObject}
+     * @param context   {@link Context}
+     * @return fields  {@link JSONArray}
+     */
+    private JSONArray loadSubFormFields(JSONObject taskValue, Context context) {
+        JSONArray fields = new JSONArray();
+        try {
+            if (taskValue != null && taskValue.has(JsonFormConstants.CONTENT_FORM)) {
+                String subFormName = taskValue.getString(JsonFormConstants.CONTENT_FORM);
+                JSONObject subForm = ContactJsonFormUtils.getSubFormJson(subFormName, "", context);
+                if (subForm.has(JsonFormConstants.CONTENT_FORM)) {
+                    fields = subForm.getJSONArray(JsonFormConstants.CONTENT_FORM);
+                }
+            }
+        } catch (JSONException e) {
+            Timber.e(e, " --> loadSubFormFields");
+        } catch (Exception e) {
+            Timber.e(e, " --> loadSubFormFields");
+        }
+        return fields;
+    }
+
+    /**
+     * Get the form title for the accordion text
+     *
+     * @param taskValue {@link JSONObject}
+     * @return title {@link String}
+     */
+    private String getFormTitle(JSONObject taskValue) {
+        String title = "";
+        if (taskValue != null && taskValue.has(JsonFormConstants.TEXT)) {
+            title = taskValue.optString(JsonFormConstants.TEXT, "");
+        }
+        return title;
+    }
+
+    /**
+     * Updates the form step1 title to match the test header
+     *
+     * @param form  {@link JSONObject}
+     * @param title {@link String}
+     */
+    private void updateFormTitle(JSONObject form, String title) {
+        try {
+            if (form != null && StringUtils.isNotBlank(title)) {
+                if (form.has(JsonFormConstants.STEP1)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=4809639)